woocommerce/woocommerce-ios
# Store creation: navigate to the progress view after necessary requests complete

Closes: #10066 


## Why

Now that we [updated the loading message](https://github.com/woocommerce/woocommerce-ios/pull/10047/) to inform the merchants that the app will remind them when the store is ready, we need to ensure the site has been created and local notification is scheduled with these two API requests before the progress UI is shown:

- `v1.1/sites/new` API request
- `/ecommerce-trial/add/ecommerce-trial-bundle-monthly` API request
- local notification is scheduled

After these are done, we can navigate to the loading screen.

## How

In `FreeTrialSummaryView`, the Continue CTA was updated from `PrimaryButtonStyle` to `PrimaryLoadingButtonStyle` so that a spinner is shown when the coordinator `StoreCreationCoordinator` is waiting for the necessary requests to be made.

These necessary requests are now in `StoreCreationCoordinator.createFreeTrialStore` that returns the site creation result async. Then, when the result is successful, it dismisses the free trial summary view, and then navigates to the in-progress view. On failure, it presents an error alert in the free trial summary view - this was an enhancement of the pre-existing error UX where the free trial summary view is dismissed, and then the store creation flow is popped to the previous screen (this error scenario probably wasn't tested before).

## Testing instructions


- Log in with WPCOM if needed
- Go to the Menu tab
- Tap on the store name to switch store, then scroll down to the bottom and tap `Add a Store`
- Tap `Create a new store`
- Continue to all the steps to create a store until the store country screen
- Turn off the internet to get an offline error
- Tap on `Continue` to launch the free trial summary view
- Tap on `Try For Free` --> an error alert should be shown
- Tap `OK` to dismiss the alert
- Turn on the internet again
- Tap on `Try For Free` --> a spinner should appear in the CTA, and then it should navigate to the in-progress UI right after seeing these logs in the console: `local_notification_scheduled`

## Screenshots


The updated error UX:


https://github.com/woocommerce/woocommerce-ios/assets/1945542/081f4662-c5a5-4c42-832b-221ab63fe51a



---
